### ISSUE TITLE
found 1 high severity vulnerability & "Hang" when working on multiple projects

### ISSUE THREAD

**Issue by matthid** [SAFE-Stack/SAFE-template#113]
I'm not sure if it was in the template but as I'm kind of new to this thing I hope someone can help anyway.
We try to evaluate the safe stack here and noticed two minor problems:

- It seems to depend on `Path            remotedev > socketcluster-client > ws` which seems to have a security issue. It suggests me to run `npm audit fix` but that doesn't seem to do anything

- When using multiple projects at the same time and keeping `fake build target Run` running in the background they will "deadlock" basically as soon as you try another instance/project (for example when having the server run and doing `dotnet fable webpack` in the same or another project). I tried playing around with the fable port but it always seems to deadlock :/

Any help is welcome also please feel free to point me to the correct place to report (or just link back yourself).

**Comment by Zaid-Ajaj** [comment]
Hi @matthid, 

> It seems to depend on Path remotedev > socketcluster-client > ws which seems to have a security issue. It suggests me to run npm audit fix but that doesn't seem to do anything

According to [this issue](https://github.com/zalmoxisus/remotedev/issues/10) the vulnerability is related to the remote debugger being subject to DDoS attacks (which I think isn't really a problem because it is a developement environment thing) 

> When using multiple projects at the same time and keeping fake build target Run running in the background they will "deadlock" basically as soon as you try another instance/project (for example when having the server run and doing dotnet fable webpack in the same or another project). I tried playing around with the fable port but it always seems to deadlock :/

Have you tried to the `--port free` option? Maybe that could help

**Comment by matthid** [comment]
> (which I think isn't really a problem because it is a developement environment thing)

Yes but I figured it should be fixed anyway as this is not a good getting-started experience. Also it will make people ignore those warnings... Additionally the dependency was not a dev-dependency on my system so there is still something to fix imho ;)

> Have you tried to the --port free option? Maybe that could help

Yes that was one of the things I tried while playing around with the ports. AFAIK it improved the situation but would still deadlock eventually. Maybe the template should be changed to use `--port free` out of the box everywhere?

**Comment by matthid** [comment]
@Zaid-Ajaj Ok thanks for bringing that up again. `--port free` seems to work as long as it is used everywhere. It definitely should be the default considering how fragile a fixed port seems to be...

**Comment by theimowski** [comment]
Indeed `remotedev` is in `dependencies`, not `devDependencies` - should it be moved?

**Comment by matthid** [comment]
Do we even need it?

**Comment by theimowski** [comment]
TBH, I have no idea - I copied that over from [BookStore](https://github.com/SAFE-Stack/SAFE-BookStore/blob/master/package.json#L9) 

Regarding `--port free` I agree this should be improved, however this is not sufficient - i.e. we still have port 8080 and 8085 hardcoded [for web server and webpack](https://safe-stack.github.io/docs/faq-build/#developing-safe-applications)

**Comment by theimowski** [comment]
I've created #114 to track multiple projects issue
Let's keep this focused on `remotedev`

**Comment by theimowski** [comment]
`remotedev` is used by https://elmish.github.io/debugger/ but it should be in `devDependencies` 

I'm still not sure how to fix the audit warning though, pinning `ws` package to 3.3.1 which is "unaffected" doesn't help really

**Comment by theimowski** [comment]
Seems that the issue won't go until `remotedev` packages updates `socketcluster-client` dependency. Pinning `ws` doesn't help here
